.github/workflows/pre-merge-checks.yml

```diff
@@ -30,8 +30,42 @@ jobs:
       - name: Run Spotless
         run: ./gradlew --no-daemon spotlessCheck
 
-  run-tests:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
```

Comment:
is this missing something at the end?

Reply:
No, this is by design. I believe this was, originally, lifted straight from the docs (https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows), it does partial matching to pick up a similar cache if the one specified in `key` doesn't exist.

Reply:
Ah nice, TIL!

Reply:
Though, actually, just spotted in the docs, we can simplify this as the `setup-java` step has it built it.